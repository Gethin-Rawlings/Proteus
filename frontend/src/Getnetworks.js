import React, { Component } from 'react';


const urlForNetworks  = network => 'http://localhost:5000/rounds'

class Getnetworks extends Component {
    constructor(props){
        super(props)
        this.state = {
            requestFailed: false
        }
    }
    componentDidMount() { 
        fetch(urlForNetworks(this.props.network)) 
        .then(response => { 
            if (!response.ok) { 
                throw Error("Network request failed") 
            } 
                 return response 
               }) 
               .then(d => d.json()) 
               
               .then(d => { 
                 this.setState({ 
                    Getnetworks: JSON.stringify(d)
                 }) 
               }, () => { 
                 this.setState({ 
                   requestFailed: true 
                 }) 
               }) 
           } 
           render() { 
             if (this.state.requestFailed) return <p>Failed!</p> 
             if (!this.state.Getnetworks) return <p>Loading...</p> 
             let returnData = JSON.parse(this.state.Getnetworks)
             if (returnData.length === 0 ) return <select id = "Network"><option>No open rounds</option></select>
             return (
                   <select id = "Network">
                     {returnData.map(p => <option value={p.org_organisation_id}>{p.org_description}</option>)}
                   </select>
             ) 
           } 
         } 
         
        export default Getnetworks; 
        