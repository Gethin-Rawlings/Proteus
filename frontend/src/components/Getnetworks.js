import React, { Component } from 'react';


const urlForNetworks  = network => 'http://172.18.0.2:5000/organisations?type=network'

class GetNetworks extends Component {
    constructor(props){
        super(props) 
        this.state = {
            requestFailed: false,
            value: ''
        }
        this.handleChange = this.handleChange.bind(this);
    }
    handleChange(event) {
      const target = event.target;
      const name = target.name;
      const value = target.value;
      this.setState({[name]: value});
    }
    
    componentDidMount() { 
        fetch(urlForNetworks(this.props.network)) 
        .then(response => { 
            if (!response.ok) { 
                throw Error("Network request failed") 
            } 
                 return response 
               }) 
               .then(d => d.json()) 
               
               .then(d => { 
                 this.setState({ 
                    Getnetworks: JSON.stringify(d)
                 }) 
               }, () => { 
                 this.setState({ 
                   requestFailed: true 
                 }) 
               }) 
           } 
           render() { 
             if (this.state.requestFailed) return <p>Failed!</p> 
             if (!this.state.Getnetworks) return <p>Loading...</p> 
             let returnData = JSON.parse(this.state.Getnetworks)
             return (
                   <select name = "network"  onChange={this.handleChange}>
                   <option disabled selected value> -- select an Network -- </option>
                     {returnData.map(p => <option  value={p.org_organisation_id}>{p.org_description}</option>)}
                   </select>
             )  
           } 
         } 
         
        export default GetNetworks; 
        