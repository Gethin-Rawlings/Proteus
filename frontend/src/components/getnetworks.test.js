import React from 'react';
import {shallow, mount} from 'enzyme';
import Getnetworks from './Getnetworks'

 test('getnetworks renders as expected' ,() => {
    const component = shallow(<Getnetworks />);
    const tree = component;
    expect(tree).toMatchSnapshot();
 })

 jest.mock("./apiCalls")
    it('should respond to change event and change the state of the networks drop down', done => {
      const spy = jest.spyOn(Getnetworks.prototype, 'handleChange')        
      const wrapper = mount(<Getnetworks />);
      
      setTimeout(()=>{
      wrapper.update();
      wrapper.find('.network').simulate('change', { target: { Getnetworks: 'data', value: 'blah@gmail.com' } });
      expect(wrapper.state('data')).toEqual('blah@gmail.com');
      expect(Getnetworks.prototype.data).toHaveBeenCalledTimes(1)
    } , done())
})
/*
const spy = jest.spyOn(Component.prototype, 'getData');
    mount(<Component />);
    expect(Component.prototype.getData).toHaveBeenCalledTimes(1)
*/