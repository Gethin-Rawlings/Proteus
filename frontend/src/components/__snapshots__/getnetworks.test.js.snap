// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getnetworks renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GetNetworks />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Loading...",
      "className": "network",
      "name": "network",
    },
    "ref": null,
    "rendered": "Loading...",
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Loading...",
        "className": "network",
        "name": "network",
      },
      "ref": null,
      "rendered": "Loading...",
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
